### DISCUSSION TITLE
Patching multiple metadata fields

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#15493] [Q&A] [answered]
Hey we're using the lastest cloud version 5.4.5

the following patch only updates the last "field" but the first one gets ignored is this a bug or a known short coming?  via studio

``` sql
from "EventDocuments" 
update {
  
    this["@metadata"]["ExpirationDate"] = "bla"
    this["@metadata"]["IsArchived"] = false
}
```

Cheers

**Comment by garayx** [comment, marked as answer]
Patch uses JavaScript engine, so I guess `this["@metadata"]` is [passed by value](https://stackoverflow.com/a/6605700/11341261), you have to assign it to a different object and then populate and reassign back to `this["@metadata"]`.

```
from "EventDocuments" 
update {
    var m = this["@metadata"];
    m["ExpirationDate"] = "bla"
    m["IsArchived"] = false
    this["@metadata"] = m;
}
```

**Reply by SbiCA** [reply]
@garayx  thx that did the trick 🎉
